tea_test.go

```diff
@@ -84,3 +96,34 @@ func TestTeaKill(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestTeaBatchMsg(t *testing.T) {
+	var buf bytes.Buffer
+	var in bytes.Buffer
+
+	inc := func() Msg {
+		return incrementMsg{}
+	}
+
+	m := &testModel{}
+	p := NewProgram(m, WithInput(&in), WithOutput(&buf))
+	go func() {
+		p.Send(batchMsg{inc, inc})
+
+		for {
+			time.Sleep(time.Millisecond)
+			if m.counter.Load() == 2 {
```

Comment:
Though this is _extremely_ unlikely to miss this condition. I usually prefer using `>=` for these types of situations so that we don't accidentally cause a infinitely for loop (albeit _extremely_ unlikely).

```suggestion
			if m.counter.Load() >= 2 {
```

Reply:
Feel free to ignore this though since it is primarily just paranoia, since it's the likelihood of this happening is negligible.

Reply:
It's a fair point actually. We could run into this situation if an `incrementMsg` gets duplicated by whatever bug. Luckily tests would still get aborted and never run indefinitely, but it's worth making the change. Thanks for the heads up!